doc/en/how-to/types.rst

```diff
@@ -0,0 +1,142 @@
+.. _types:
+
+How to use Types with Pytest
+===========================================
+
+You can add typing in pytest functions that helps to specify what type of data a function expects and returns.
+This guide explains how to apply typing in pytest to improve code readability and prevent type-related errors.
+
+
+1. Introduction to Typing
+-----------------
+
+Typing in Python helps developers specify the expected data types of variables, function parameters, and return values.
+It improves code clarity and prevents type errors.
+
+For example:
+
+.. code-block:: python
+
+    def add(x: int, y: int) -> int:
+        return x + y
+
+In this example:
+`x: int` and `y: int` mean that `x` and `y` should be integers.
+`-> int` shows that the function returns an integer.
+
+It makes the intentions clear that both the input and output are expected to be integers.
+
+2. Typing Test Functions
+-----------------
+Test functions in pytest check whether the code runs correctly.
+While test functions do not return values, we can add `-> None` as the return type.
+
+For example:
+
+.. code-block:: python
+
+    import pytest
+
+
+    def test_add() -> None:
+        result = add(2, 3)
+        assert result == 5
+
+In this function:
+`test_add` is typed as `-> None` because it does not return anything.
```

Comment:
We can instead mention that while at first marking a function as returning `-> None` might seem useless, having at least a type annotation means most type checkers will now type check the function, so it is useful because of that.